src/FilenameParsing/FileIDHelperResolutionStrategy.php

```diff
@@ -118,7 +118,7 @@ private function resolveWithHash(ParsedFileID $parsedFileID)
         /** @var File $file */
         $file = Versioned::withVersionedMode(function () use ($parsedFileID) {
             Versioned::set_stage($this->getVersionedStage());
-            return File::get()->filter(['FileFilename' => $parsedFileID->getFilename()])->first();
+            return File::get()->filter(['FileFilename:case' => $parsedFileID->getFilename()])->first();
```

Comment:
Changes in this file ar not really related to this PR. I realised that those check were getting confused in some circumstances because the Filename check was not case sensitive.